x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_exploration/components/exploration_results_table/use_exploration_results.ts

```diff
@@ -73,8 +73,15 @@ export const useExplorationResults = (
     dataGrid.resetPagination();
   }, [JSON.stringify(searchQuery)]);
 
+  // The pattern using `didCancel` allows us to abort out of date remote request.
+  // We wrap `didCancel` in a object so we can mutate the value as it's being
+  // passed on to `getIndexData`.
   useEffect(() => {
-    getIndexData(jobConfig, dataGrid, searchQuery);
+    const options = { didCancel: false };
+    getIndexData(jobConfig, dataGrid, searchQuery, options);
+    return () => {
+      options.didCancel = true;
```

Comment:
shall we create a custom hook, let's call it `useSafeEffect` for instance, that wraps the callback with `didCancel` check under the hood?

Reply:
yep that's something we could do in a follow up for 7.11 - for this 7.10 fix I wanted to touch as little code as possible that's why I went for duplicate code and no new utils. i'd also like to refactor `getIndexData` a bit so we don't have to use the mutable object but just a plain variable.